### PR TITLE
feat(graphql): use private key in session

### LINKED ISSUE #316: Organize login session

### PR DESCRIPTION
**Pull request by moreal**
It resolves #316.

This pull request makes it manage private key by session (i.e. user), with key store as account backend.

Since this pull request, it will have [Moq](https://github.com/moq/moq4) to mock services, as a dependency.

### CODE REVIEW

File: NineChronicles.Headless.Tests/SessionTest.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            Assert.Contains(bar, _session.Keys);
+            Assert.Contains(session, _session.Keys);
+            _session.Remove(bar);
+            Assert.DoesNotContain(foo, _session.Keys);
+            Assert.DoesNotContain(bar, _session.Keys);
+            Assert.Contains(session, _session.Keys);
+            _session.Remove(session);
+            Assert.DoesNotContain(foo, _session.Keys);
+            Assert.DoesNotContain(bar, _session.Keys);
+            Assert.DoesNotContain(session, _session.Keys);
```
Comment (ipdae) [inline NineChronicles.Headless.Tests/SessionTest.cs:47]:
```suggestion
            Assert.Empty(_session.Keys);
```

**Review by ipdae — APPROVED** [review]

**Review by area363 — APPROVED** [review]

**Comment by area363** [comment]
There's a conflict with `NineChronicles.Headless/Controllers/GraphQLController.cs`.

**Comment by moreal** [comment]
There were added new changes with rebasing:

- cb0a720 login session with raw private key, for planetarium/9c-launcher flow
- fe085e2 broadcast all events related to addresses in key store to subscriber, for planetarium/9c-launcher flow
- e39e69d add missing `Notification.Receiver` field to make users able to filter

Please review again @planetarium/9c-backend 🙏🏻

**Review by ipdae — APPROVED** [review]

File: NineChronicles.Headless/Controllers/GraphQLController.cs [hunk 2]
```diff
@@ -241,66 +245,61 @@ bool NeedsRefillNotification(AvatarState avatarState)
...
-            if (StandaloneContext.NineChroniclesNodeService is null)
-            {
-                throw new InvalidOperationException(
-                    $"{nameof(StandaloneContext.NineChroniclesNodeService)} is null.");
-            }
-
-            if (StandaloneContext.NineChroniclesNodeService.MinerPrivateKey is null)
+            List<Tuple<Guid, ProtectedPrivateKey>>? tuples =
+                StandaloneContext.KeyStore?.List().ToList();
+            if (tuples is null || !tuples.Any())
```
[the remark this answers is not in the document]
Comment (moreal) [reply NineChronicles.Headless/Controllers/GraphQLController.cs:263]: It is unable to compile with the message, `Cannot implicitly convert type 'System.Nullable<bool>' to 'bool'`.

**Pull request merged**